scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -317,6 +318,26 @@
     default=True,
 )
 
+define_config_option(
+    __monitor__,
+    "tcp_request_parser",
+    "Optional (defaults to default). Which TCP packet data request parser to use. Most users "
+    "should leave this as is.",
+    default="default",
+    convert_to=six.text_type,
+)
+
+define_config_option(
+    __monitor__,
+    "tcp_incomplete_frame_timeout",
+    "How long we wait (in seconds) for a complete frame / syslog message when running in TCP mode "
+    "with batched request parser before giving up and flushing what has accumulated in the buffer.",
+    default=60,
```

Comment:
I would do 5 seconds as the default.  We do something similar in the main part of the agent (we wait for a new line to send an `/addEvents` request, so we timeout after 5 seconds if we've gotten a partial line).  We haven't had any complaints about that to date.